settings.gradle

```
@@ -60,10 +60,12 @@ if (System.env['CI'] != null) {
 }
 
 gradleEnterprise {
-	buildScan {
-		termsOfServiceUrl = "https://gradle.com/terms-of-service"
-		termsOfServiceAgree = "yes"
-		publishAlways()
+	if (System.env['CI'] != null) {
+		buildScan {
+			termsOfServiceUrl = "https://gradle.com/terms-of-service"
+			termsOfServiceAgree = "yes"
+			publishAlways()
+		}
```

```suggestion
	buildScan {
		termsOfServiceUrl = "https://gradle.com/terms-of-service"
		termsOfServiceAgree = "yes"
		publishAlwaysIf(providers.environmentVariable("CI").isPresent())
	}
```

---

All the better for configuration cache!

I've seen and used that syntax for the Kotlin DSL, but I believe the following is supposed to be more idiomatic for Groovy DSL. Which do you prefer?

```suggestion
	buildScan {
		termsOfServiceUrl = "https://gradle.com/terms-of-service"
		termsOfServiceAgree = "yes"
		publishAlwaysIf(providers.environmentVariable("CI").present)
	}
```

---

I'm happy to accept yours.